x-pack/plugins/cloud_security_posture/server/fleet_integration/fleet_integration.test.ts

```diff
@@ -5,166 +5,39 @@
-    await removeCspRulesInstancesCallback(
-      mockDeletePackagePolicy[0],
-      savedObjectRepositoryMock,
-      logger
-    );
-
-    expect(savedObjectRepositoryMock.find.mock.calls[0][0]).toMatchObject({ perPage: 10000 });
-  });
-
-  it('get default integration type from inputs with multiple enabled types', () => {
-    const mockPackagePolicy = createPackagePolicyMock();
-
-    // Both enabled falls back to default
-    mockPackagePolicy.inputs = [
-      { type: 'cloudbeat/cis_k8s', enabled: true, streams: [] },
-      { type: 'cloudbeat/cis_eks', enabled: true, streams: [] },
-    ];
-    const type = getBenchmarkInputType(mockPackagePolicy.inputs);
-    expect(type).toMatch('cis_k8s');
-  });
-
-  it('get default integration type from inputs without any enabled types', () => {
-    const mockPackagePolicy = createPackagePolicyMock();
-
-    // None enabled falls back to default
-    mockPackagePolicy.inputs = [
-      { type: 'cloudbeat/cis_k8s', enabled: false, streams: [] },
-      { type: 'cloudbeat/cis_eks', enabled: false, streams: [] },
-    ];
-    const type = getBenchmarkInputType(mockPackagePolicy.inputs);
-    expect(type).toMatch('cis_k8s');
-  });
-
-  it('get EKS integration type', () => {
-    const mockPackagePolicy = createPackagePolicyMock();
-
-    // Single EKS selected
-    mockPackagePolicy.inputs = [
-      { type: 'cloudbeat/cis_eks', enabled: true, streams: [] },
-      { type: 'cloudbeat/cis_k8s', enabled: false, streams: [] },
-    ];
-    const typeEks = getBenchmarkInputType(mockPackagePolicy.inputs);
-    expect(typeEks).toMatch('cis_eks');
-  });
-
-  it('get Vanilla K8S integration type', () => {
-    const mockPackagePolicy = createPackagePolicyMock();
-
-    // Single k8s selected
-    mockPackagePolicy.inputs = [
-      { type: 'cloudbeat/cis_eks', enabled: false, streams: [] },
-      { type: 'cloudbeat/cis_k8s', enabled: true, streams: [] },
-    ];
-    const typeK8s = getBenchmarkInputType(mockPackagePolicy.inputs);
-    expect(typeK8s).toMatch('cis_k8s');
-  });
+import { createPackagePolicyServiceMock } from '@kbn/fleet-plugin/server/mocks';
+import { ListResult, PackagePolicy } from '@kbn/fleet-plugin/common';
+import { savedObjectsClientMock } from '@kbn/core-saved-objects-api-server-mocks';
+import { isCspPackagePolicyInstalled } from './fleet_integration';
+import { loggingSystemMock } from '@kbn/core-logging-server-mocks';
+import { createPackagePolicyMock } from '@kbn/fleet-plugin/common/mocks';
+
+describe('is Csp package installed tests', () => {
+  const logger = loggingSystemMock.createLogger();
+  const soClient = savedObjectsClientMock.create();
+  const packagePolicyService = createPackagePolicyServiceMock();
+
+  beforeEach(() => jest.clearAllMocks());
+  it.each`
+    total | items                          | expectedCspPolicyResponse
+    ${1}  | ${[createPackagePolicyMock()]} | ${true}
+    ${0}  | ${[]}                          | ${false}
+  `(
```

Comment:
personally i feel like a plain for loop or calling `each` with array would be much nicer, this feels a bit alien way to iterate. 

(a good example IMO for [tagged templates](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates) is for handling a simple DSL. we use it a lot for CSS)